### ISSUE TITLE
CLI not working when updgrading to yarn berry

### ISSUE THREAD

**Issue by harrismcc** [supabase/cli#1380]
**Describe the bug**
I'm upgrading to yarn v2 (berry) but it's breaking the supabase CLI. It works normally with yarn v1

**To Reproduce**
1. Install supabase cli using yarn v1 (I'm on v1.22.19)
2. Verify commands like `supabase start` work
3. run `yarn set version berry` to upgrade, then `yarn` to re-install packages
4. Run `supabase start` again and observe error

**Expected behavior**
Supabase CLI should work properly with yarn v2

**Desktop (please complete the following information):**
 - OS: macOS
 - Version of CLI 1.86.1
 - Version of Node.js (if applicable) 18.17.0

**Additional context**
This is the full error message:
```
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for /Users/harris/Documents/untitled-notes-app/node_modules/supabase/bin/supabase. Loading extensionless files is not supported inside of "type":"module" package.json contexts. The package.json file /Users/harris/Documents/untitled-notes-app/node_modules/supabase/package.json caused this "type":"module" context. Try changing /Users/harris/Documents/untitled-notes-app/node_modules/supabase/bin/supabase to have a file extension. Note the "bin" field of package.json can point to a file with an extension, for example {"type":"module","bin":{"supabase":"./bin/supabase.js"}}
    at new NodeError (node:internal/errors:405:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:79:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:124:36)
    at defaultLoad (node:internal/modules/esm/load:84:20)
    at nextLoad (node:internal/modules/esm/loader:163:28)
    at ESMLoader.load (node:internal/modules/esm/loader:603:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:457:22)
    at new ModuleJob (node:internal/modules/esm/module_job:64:26)
    at #createModuleJob (node:internal/modules/esm/loader:480:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:434:34) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

Node.js v18.17.0
```

**Comment by sweatybridge** [comment]
Hmm I ran the following but can't seem to reproduce this issue.

```bash
$ docker run --rm --workdir=/tmp -it node:slim bash
root@8353ad017520:/tmp# yarn set version berry
➤ YN0000: Retrieving https://repo.yarnpkg.com/3.6.4/packages/yarnpkg-cli/bin/yarn.js
➤ YN0000: Saving the new release in .yarn/releases/yarn-3.6.4.cjs
➤ YN0000: Done in 0s 448ms
root@8353ad017520:/tmp# yarn init -y
{
  name: 'tmp',
  packageManager: 'yarn@3.6.4'
}
Internal Error: Process git failed to spawn
    at ChildProcess.<anonymous> (/tmp/.yarn/releases/yarn-3.6.4.cjs:392:5563)
    at ChildProcess.emit (node:events:514:28)
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
root@8353ad017520:/tmp# yarn add -D supabase
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed in 2s 869ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ supabase@npm:1.104.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ tar@npm:6.2.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ web-streams-polyfill@npm:3.2.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ write-file-atomic@npm:5.0.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 2s 57ms
➤ YN0000: ┌ Link step
➤ YN0000: │ ESM support for PnP uses the experimental loader API and is therefore experimental
➤ YN0007: │ supabase@npm:1.104.1 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 2s 854ms
➤ YN0000: Done with warnings in 7s 794ms
root@8353ad017520:/tmp# yarn supabase -h
Supabase CLI 1.104.1
...
```

Have you tried deleting any local cache and the yarn.lock files?

**Comment by sweatybridge** [comment]
I believe this has been addressed in latest yarn@3.6.4 https://github.com/yarnpkg/berry/pull/4289

feel free to reopen if it's still causing problems
